packages/react-core/src/core/providers/thirdweb-provider.tsx

```diff
@@ -0,0 +1,188 @@
+import { SupportedWallet } from "../types/wallet";
+import { showDeprecationWarning } from "../utils";
+import { ThirdwebThemeContext } from "./theme-context";
+import { ThirdwebWalletProvider } from "./thirdweb-wallet-provider";
+import { QueryClient } from "@tanstack/react-query";
+import { Chain, defaultChains } from "@thirdweb-dev/chains";
+import type { SDKOptions } from "@thirdweb-dev/sdk";
+import type { ThirdwebStorage } from "@thirdweb-dev/storage";
+import type { CreateAsyncStorage } from "@thirdweb-dev/wallets";
+import React, { useMemo } from "react";
+
+/**
+ * The possible props for the ThirdwebProvider.
+ */
+export interface ThirdwebProviderProps<
+  TChains extends Chain[] = typeof defaultChains,
+> {
+  /**
+   * The network to use for the SDK.
+   */
+  activeChain?: TChains[number]["chainId"] | TChains[number]["slug"] | Chain;
+
+  /**
+   * Chains to support. If not provided, will default to the chains supported by the SDK.
+   */
+  supportedChains?: TChains;
+
+  /**
+   * The {@link SDKOptions | Thirdweb SDK Options} to pass to the thirdweb SDK
+   * comes with sensible defaults
+   */
+  sdkOptions?: SDKOptions;
+
+  /**
+   * An array of wallets that the dApp supports
+   * If not provided, will default to Metamask (injected), Coinbase wallet and Device wallet
+   *
+   * @example
+   * You can Import the wallets you want to support from `@thirdweb-dev/wallets` and pass them to `supportedWallets`
+   *
+   * ```jsx title="App.jsx"
+   * import { ThirdwebProvider } from "@thirdweb-dev/react";
+   *
+   * const App = () => {
+   *   return (
+   *     <ThirdwebProvider>
+   *       <YourApp />
+   *     </ThirdwebProvider>
+   *   );
+   * };
+   * ```
+   */
+  supportedWallets: SupportedWallet[];
+
+  /**
+   * Metadata to pass to wallet connect and walletlink wallet connect. (Used to show *which* dApp is being connected to in mobile wallets that support it)
+   * Defaults to just the name being passed as `thirdweb powered dApp`.
+   */
+  dAppMeta?: DAppMetaData;
+
+  /**
+   * The configuration used for thirdweb auth usage. Enables users to login
+   * to backends with their wallet.
+   */
+  authConfig?: ThirdwebAuthConfig;
+
+  /**
+   * The storage interface to use with the sdk.
+   */
+  storageInterface?: ThirdwebStorage;
+
+  /**
+   * The react-query client to use. (Defaults to a default client.)
+   */
+  queryClient?: QueryClient;
+
+  /**
+   * Whether or not to attempt auto-connect to a wallet.
+   */
+  autoConnect?: boolean;
+
+  // api keys that can be passed
+  thirdwebApiKey?: string;
+  alchemyApiKey?: string;
+  infuraApiKey?: string;
+
+  /**
+   * A partial map of chainIds to rpc urls to use for certain chains
+   * If not provided, will default to the rpcUrls of the chain objects for the supported chains
+   * @deprecated - use `chains` instead
+   */
+  chainRpc?: Record<number, string>;
+
+  theme?: "light" | "dark";
+
+  createWalletStorage: CreateAsyncStorage;
```

Comment:
is this required? We should probably default it to the browser option (since tecnically people could use this directly stil)

Reply:
this is required in `react-core`, but not in `react` or `react-native`
they export their own version of `ThirdwebProvider` - where `createWalletStorage` is omitted from props.

react/src/evm/providers/thirdweb-provider.tsx :
 
<img width="808" alt="image" src="https://user-images.githubusercontent.com/22043396/224246750-f2d06324-22e7-474c-99df-59a595c2943a.png">